Content.Shared/Access/Components/IdCardComponent.cs

```diff
@@ -24,7 +24,7 @@ public sealed partial class IdCardComponent : Component
     /// <summary>
     /// The state of the job icon rsi.
     /// </summary>
-    [DataField("jobIcon", customTypeSerializer: typeof(PrototypeIdSerializer<StatusIconPrototype>))]
+    [DataField("jobIcon", customTypeSerializer: typeof(PrototypeIdSerializer<JobIconPrototype>))]
     [AutoNetworkedField]
     public string JobIcon = "JobIconUnknown";
```

Comment:
Just make it use protoid

Reply:
Done. I had to remove a few null checks for JobIcon my compiler was complaining about after that (see latest commit), but since it cannot be null and has a default value it should be fine.